File: RQ2_Prompt_Elements/OpenAI_Data/HumanEvalPython_output/scenario2/id_105T2000Test.py
# id_105_Test.java
from scenario2.id_105 import by_length

import pytest

class id_105_Test:
    """
    Test class of id_105.py.
    It contains ten unit test cases for the :func:`scenario2.id_105.by_length`.
    """


    def test_by_length_1(self):
        """
        Test case 1 for :func:`scenario2.id_105.by_length`.
        """
        assert by_length([1, 2, 3, 4, 5, 6, 7, 8, 9]) == ["Nine", "Eight", "Seven", "Six", "Five", "Four", "Three", "Two", "One"]

    def test_by_length_2(self):
        """
        Test case 2 for :func:`scenario2.id_105.by_length`.
        """
        assert by_length([1, 2, 3, 4, 5, 6, 7, 8, 9, 10]) == ["Nine", "Eight", "Seven", "Six", "Five", "Four", "Three", "Two", "One"]

    def test_by_length_3(self):
        """
        Test case 3 for :func:`scenario2.id_105.by_length`.
        """
        assert by_length([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]) == ["Nine", "Eight", "Seven", "Six", "Five", "Four", "Three", "Two", "One"]

    def test_by_length_4(self):
        """
        Test case 4 for :func:`scenario2.id_105.by_length`.
        """
        assert by_length([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]) == ["Nine", "Eight", "Seven", "Six", "Five", "Four", "Three", "Two", "One"]

    def test_by_length_5(self):
        """
        Test case 5 for :func:`scenario2.id_105.by_length`.
        """
        assert by_length([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13]) == ["Nine", "Eight", "Seven", "Six", "Five", "Four", "Three", "Two", "One"]

    def test_by_length_6(self):
        """
        Test case 6 for :func:`scenario2.id_105.by_length`.
        """
        assert by_length([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14]) == ["Nine", "Eight", "Seven", "Six", "Five", "Four", "Three", "Two", "One"]

    def test_by_length_7(self):
        """
        Test case 7 for :func:`scenario2.id_105.by_length`.
        """
        assert by_length([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15]) == ["Nine", "Eight", "Seven", "Six", "Five", "Four", "Three", "Two", "One"]

    def test_by_length_8(self):
        """
        Test case 8 for :func:`scenario2.id_105.by_length`.
        """
        assert by_length([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16]) == ["Nine", "Eight", "Seven", "Six", "Five", "Four", "Three", "Two", "One"]

    def test_by_length_9(self):
        """
        Test case 9 for :func:`scenario2.id_105.by_length`.
        """
        assert by_length([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17]) == ["Nine", "Eight", "Seven", "Six", "Five", "Four", "Three", "Two", "One"]

    def test_by_length_10(self):
        """
        Test case 10 for :func:`scenario2.id_105.by_length`.
        """
        assert by_length([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18]) == ["Nine", "Eight", "Seven", "Six", "Five", "Four", "Three", "Two", "One"]

File: RQ2_Prompt_Elements/StarCoder_Data/HumanEvalJava_output/scenario2/id_105T2000Test.java
// ByLengthTest.java
package scenario2;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

/**
* Test class of {@link ByLength}.
* It contains ten unit test cases for the {@link ByLength#byLength(List)} method.
*/
class ByLengthTest {

    /**
     * Test case for the {@link ByLength#byLength(List)} method.
     * It tests the case where the input array is null.
     */
    @Test
    void testByLengthNull() {
        List<Object> arr = null;
        List<Object> result = ByLength.byLength(arr);
        assertEquals(null, result);
    }

    /**
     * Test case for the {@link ByLength#byLength(List)} method.
     * It tests the case where the input array is empty.
     */
    @Test
    void testByLengthEmpty() {
        List<Object> arr = new ArrayList<Object>();
        List<Object> result = ByLength.byLength(arr);
        assertEquals(0, result.size());
    }

    /**
     * Test case for the {@link ByLength#byLength(List)} method.
     * It tests the case where the input array contains only integers.
     */
    @Test
    void testByLengthIntegers() {
        List<Object> arr = new ArrayList<Object>();
        arr.add(1);
        arr.add(2);
        arr.add(3);
        arr.add(4);
        arr.add(5);
        arr.add(6);
        arr.add(7);
        arr.add(8);
        arr.add(9);
        List<Object> result = ByLength.byLength(arr);
        assertEquals(9, result.size());
        assertEquals("One", result.get(0));
        assertEquals("Two", result.get(1));
        assertEquals("Three", result.get(2));
        assertEquals("Four", result.get(3));
        assertEquals("Five", result.get(4));
        assertEquals("Six", result.get(5));
        assertEquals("Seven", result.get(6));
        assertEquals("Eight", result.get(7));
        assertEquals("Nine", result.get(8));
    }

    /**
     * Test case for the {@link ByLength#byLength(List)} method.
     * It tests the case where the input array contains only strings.
     */
    @Test
    void testByLengthStrings() {
        List<Object> arr = new ArrayList<Object>();
        arr.add("One");
        arr.add("Two");
        arr.add("Three");
        arr.add("Four");
        arr.add("Five");
        arr.add("Six");
        arr.add("Seven");
        arr.add("Eight");
        arr.add("Nine");
        List<Object> result = ByLength.byLength(arr);
        assertEquals(9, result.size());
        assertEquals("One", result.get(0));
        assertEquals("Two", result.get(1));
        assertEquals("Three", result.get(2));
        assertEquals("Four", result.get(3));
        assertEquals("Five", result.get(4));
        assertEquals("Six", result.get(5));
        assertEquals("Seven", result.get(6));
        assertEquals("Eight", result.get(7));
        assertEquals("Nine", result.get(8));
    }

    /**
     * Test case for the {@link ByLength#byLength(List)} method.
     * It tests the case where the input array contains both integers and strings.
     */
    @Test
    void testByLengthMixed() {
        List<Object> arr = new ArrayList<Object>();
        arr.add(1);
        arr.add("Two");
        arr.add(3);
        arr.add("Four");
        arr.add(5);
        arr.add("Six");
        arr.add(7);
        arr.add("Eight");
        arr.add(9);
        List<Object> result = ByLength.byLength(arr);
        assertEquals(9, result.size());
        assertEquals("One", result.get(0));
        assertEquals("Two", result.get(1));
        assertEquals("Three", result.get(2));
        assertEquals("Four", result.get(3));
        assertEquals("Five", result.get(4));
        assertEquals("Six", result.get(5));
        assertEquals("Seven", result.get(6));
        assertEquals("Eight", result.get(7));
        assertEquals("Nine", result.get(8));
    }

    /**
     * Test case for the {@link ByLength#byLength(List)} method.
     * It tests the case where the input array contains integers and strings of different lengths.
     */
    @Test
    void testByLengthMixedDifferentLengths() {
        List<Object> arr = new ArrayList<Object>();
        arr.add(1);
        arr.add("Two");
        arr.add(3);
        arr.add("Four");
        arr.add(5);
        arr.add("Six");
        arr.add(7);
        arr.add("Eight");
        arr.add(9);
        arr.add("Ten");
        List<Object> result = ByLength.byLength(arr);
        assertEquals(10, result.size());
        assertEquals("One", result.get(0));
        assertEquals("Two", result.get(1));
        assertEquals("Three", result.get(2));
        assertEquals("Four", result.get(3));
        assertEquals("Five", result.get(4));
        assertEquals("Six", result.get(5));
        assertEquals("Seven", result.get(6));
        assertEquals("Eight", result.get(7));
        assertEquals("Nine", result.get(8));
        assertEquals("Ten", result.get(9));
    }
}
